astropy/modeling/core.py

```diff
@@ -34,61 +32,55 @@
 from astropy.units import Quantity, UnitsError, dimensionless_unscaled
 from astropy.units.utils import quantity_asanyarray
 from astropy.utils import (sharedmethod, find_current_module,
-                           InheritDocstrings, OrderedDescriptorContainer,
+                           InheritDocstrings,
                            check_broadcast, IncompatibleShapeError, isiterable)
 from astropy.utils.codegen import make_function_with_signature
 from astropy.utils.exceptions import AstropyDeprecationWarning
+from astropy.utils.misc import get_parameters
 from .utils import (combine_labels, make_binary_operator_eval,
-                    ExpressionTree, AliasDict, get_inputs_and_params,
+                    AliasDict, get_inputs_and_params,
                     _BoundingBox, _combine_equivalency_dict)
 from astropy.nddata.utils import add_array, extract_array
+from .parameters import (Parameter, InputParameterError,
+                         param_repr_oneline, _tofloat)
 
-from .parameters import Parameter, InputParameterError, param_repr_oneline
+from collections import deque
 
 
 __all__ = ['Model', 'FittableModel', 'Fittable1DModel', 'Fittable2DModel',
-           'custom_model', 'ModelDefinitionError']
-
-
-class ModelDefinitionError(TypeError):
-    """Used for incorrect models definitions"""
+           'CompoundModel', 'custom_model', 'ModelDefinitionError']
 
 
 def _model_oper(oper, **kwargs):
     """
-    Returns a function that evaluates a given Python arithmetic operator
-    between two models.  The operator should be given as a string, like ``'+'``
-    or ``'**'``.
-
-    Any additional keyword arguments passed in are passed to
-    `_CompoundModelMeta._from_operator`.
+    This is an alternate version of compound models intended to use
+    much less memory than the default.
```

Comment:
The old docstring was more informative, in particular the first paragraph. Also is this still an alternative, or just the main version now?